tracer/test/Datadog.Trace.Tests/TracerSettingsTests.cs

```diff
@@ -99,7 +99,7 @@ public void ReplaceLocalhost(string original, string expected)
                 { ConfigurationKeys.AgentUri, original }
             };
 
-            var tracerSettings = new TracerSettings(new NameValueConfigurationSource(settings));
+            var tracerSettings = new ImmutableTracerSettings(new TracerSettings(new NameValueConfigurationSource(settings)));
```

Comment:
I'm a little on the fence, but I feel like this test belongs in `ImmutableExporterSettingsTests` since that is where the replacement of localhost occurs